### ISSUE TITLE
Ordering of placement_strategy 

### ISSUE THREAD

**Issue by atsushi-ishibashi** [terraform-providers/terraform-provider-aws#1466]
How about supporting the order of placement strategy?
https://github.com/terraform-providers/terraform-provider-aws/blob/48fdbc0997746c9c226d36bf358d8884b176c0da/aws/resource_aws_ecs_service.go#L106
According to [aws docs](https://docs.aws.amazon.com/ja_jp/AmazonECS/latest/APIReference/API_CreateService.html), maybe the order is associated with that of `"placementStrategy"` list.

I'd like to configure like AZ Balance Spread provided by AWS console so I wrote the codes below. But placement_strategy is set as TypeSet, so I couldn't create placement_strategy of AZ Balance Spread.
```
resource "aws_ecs_service" "hoge" {
  placement_strategy {
    type  = "spread"
    field = "instanceId"
  }

  placement_strategy {
    type  = "spread"
    field = "attribute:ecs.availability-zone"
  }
}
```

So I propose that adding a `order` key to `placement_strategy` or applying TypeList to placement strategy.

**Comment by atsushi-ishibashi** [comment]
> maybe the order is associated with that of "placementStrategy" list.
So far I'm not assure that the specification of placement strategy is as above.
I'm researching...

**Comment by atsushi-ishibashi** [comment]
My assumption seems to be correct.
> The following strategy distributes tasks evenly across Availability Zones and then distributes tasks evenly across the instances within each Availability Zone.

http://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-placement-strategies.html

**Comment by dtserekhman-starz** [comment]
Any update on this ticket of when the ordering can be specified? Otherwise, it appears the ordering is arbitrary, and Terraform does not follow the order in which placement_strategy attributed are specified.

**Comment by nathanielks** [comment]
@dtserekhman-starz see https://github.com/terraform-providers/terraform-provider-aws/pull/1476

**Comment by bflad** [comment]
In order to support this change without breaking compatibility, we have implemented a new argument `ordered_placement_strategy` and deprecated the existing `placement_strategy` argument so you can opt into the new behavior. This will be released with v1.17.0 of the AWS provider, likely in two days. We will remove `placement_strategy` in the next major version of the AWS provider.
